service/common/src/main/java/org/apache/polaris/service/events/PolarisEventListener.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.polaris.service.events;
+
+/**
+ * Represents an event listener that can respond to notable moments during Polaris's execution.
+ * Users can either extend this interface and implement handlers for all events or, for ease, extend
+ * DefaultPolarisEventListener and only have to handle a subset of events. Event details are
+ * documented under the event objects themselves.
+ */
+public interface PolarisEventListener {
```

Comment:
I wonder if we really need the `Polaris` prefix here. 

Also, unfortunately, I think we need javadocs on the methods.

Reply:
Yea we could get rid of the Polaris prefix.

WDYT about `@link` javadocs that point to the actual event data class? I just didn't wanna duplicate the comment in 2 places. Spark seems to have fairly vague comments on the methods that don't provide much more info than the method name itself, but also Spark doesn't seem to have the detailed comments on the event data classes themselves.

Reply:
Yeah that's fair, let's just add `@link`s

Reply:
The "Polaris" prefix would deserve a discuss thread imho. Many components have this prefix, but not all.

Reply:
Many have it, and many don't -- agreed that we should standardize and agreed that a discussion thread is a good place to do that.

For this individual decision on this PR, I'm not sure a decision either way would require a huge discussion given the current lack of standardization.